projects/github-actions/test-results-to-slack/src/github.js

```diff
@@ -0,0 +1,230 @@
+const github = require( '@actions/github' );
+const extras = require( './extra-context' );
+
+/**
+ * Decides if the current workflow failed
+ *
+ * @param {string} token - GitHub token
+ */
+async function isWorkflowFailed( token ) {
+	// eslint-disable-next-line new-cap
+	const octokit = new github.getOctokit( token );
+	const {
+		payload: { repository },
+		runId,
+	} = github.context;
+
+	// Get the list of jobs for the current workflow run
+	const response = await octokit.rest.actions.listJobsForWorkflowRun( {
+		owner: repository.owner.login,
+		repo: repository.name,
+		run_id: runId,
+	} );
+
+	// Get unique list of conclusions of completed jobs
+	const conclusions = [
+		...new Set(
+			response.data.jobs.filter( job => job.status === 'completed' ).map( job => job.conclusion )
+		),
+	];
+
+	// Decide if any we'll treat this run as failed
+	return !! conclusions.some( conclusion => conclusion !== 'success' );
+}
+
+/**
+ * Returns an object with notification data.
+ * Properties: `text` for notification's text and `id` for a unique identifier for the message.
+ * that can be used later on to find this message and update it or send replies.
+ *
+ * @param {boolean} isFailure - whether the workflow is failed or not
+ */
+async function getNotificationData( isFailure ) {
+	const {
+		context: { eventName, sha, payload, runId, actor, serverUrl },
+	} = github;
+	const { refType, refName, runAttempt, triggeringActor, repository } = extras;
+	let target = `for ${ sha }`;
+	let msgId;
+	const contextElements = [];
+	const buttons = [];
+	const style = isFailure ? 'danger' : 'primary';
+
+	if ( eventName === 'pull_request' ) {
```

Comment:
Do you think it would make sense to extract these conditionals into their own functions?

Reply:
It might make sense. I'll look into this in the next PR.